### DISCUSSION TITLE
Which Pro / Enterprise features have bad ActiveJob interactions?

### DISCUSSION THREAD

**Discussion by lorrin** [sidekiq/sidekiq#5738] [Q&A] [answered]
The [docs](https://github.com/mperham/sidekiq/wiki/Active-Job#commercial-features) say _"Some of Sidekiq Pro and Sidekiq Enterprise's features will break in unpredictable ways if you try to use Active Job with those features."_. But it only mentions Batch and ActiveJob retry as having a bad interaction. Can I get more information on areas of concern? Our existing code uses ActiveJob extensively and we're considering migrating to Sidekiq Enterprise. I'm hoping to get going with Sidekiq without having to change all ActiveJob related code up-front.

**Comment by mperham** [comment, marked as answer]
This is nebulous for a reason: we can't possibly enumerate all things which are safe vs dangerous. Features can change and versions can contain fixes over time.

I believe you will find that most things are safe but there are major points of divergence; for instance, ActiveJob implements its own retry behavior which is invisible to Sidekiq. If you don't use ActiveJob retries and use Sidekiq native retries instead, that's 90% of the incompatibilities AFAIK. `sidekiq_retries_exhausted` doesn't work with AJs but you can use a global :death handler instead, etc.
